src/components/WidgetWindow.tsx

```diff
@@ -100,7 +100,13 @@ const StyledCloseButton = styled.button`
 const WidgetWindow = ({ children }: { children: React.ReactNode }) => {
   const { isOpen, setIsOpen } = useWidgetOpen();
   const [isExpanded, setIsExpanded] = useState(false);
-  const { customUserAgentParam } = useConstantState();
+  const { customUserAgentParam, onWidgetExpandStateChange } =
+    useConstantState();
+
+  const onExpandButtonToggle = () => {
+    onWidgetExpandStateChange?.(!isExpanded);
+    setIsExpanded((prev) => !prev);
+  };
```

Comment:
Note that below solution redundantly calls twice
https://sendbird.slack.com/archives/C024Q3BGCRL/p1716261091908299

Reply:
turned out react strictmode calls hook twice.

Reply:
Since strictmode is used in dev only, at production there wont be redundancy issue with above solution. Changed to above solution.